terraform/lib/dependabot/terraform.rb

```diff
@@ -17,3 +17,6 @@
 require "dependabot/dependency"
 Dependabot::Dependency.
   register_production_check("terraform", ->(_) { true })
+
+require "dependabot/utils"
+Dependabot::Utils.register_always_clone("terraform")
```

Comment:
I'm pretty sure this is all that needs to be done, in addition to the file updater change to call `SharedHelpers.in_a_temporary_repo_directory`. Tested with a dry-run and looks like the updater checks this flag to turn on cloning.

Reply:
Any reason we shouldn't be doing this?

Reply:
I think this is it! 👍